src/huggingface_hub/inference/_client.py

```diff
@@ -1166,19 +1156,8 @@ def image_classification(
         ```
         """
         parameters = {"function_to_apply": function_to_apply, "top_k": top_k}
-
-        if all(parameter is None for parameter in parameters.values()):
-            data = image
-            payload: Optional[Dict[str, Any]] = None
-
-        else:
-            data = None
-            payload = {"inputs": _b64_encode(image)}
-            for key, value in parameters.items():
-                if value is not None:
-                    payload.setdefault("parameters", {})[key] = value
-
-        response = self.post(json=payload, data=data, model=model, task="image-classification")
+        payload = self._prepare_payload(image, parameters=parameters)
+        response = self.post(json=payload.json, data=payload.raw_data, model=model, task="image-classification")
```

Comment:
Typically here (based on suggestions above), it would lead to something like this:

```suggestion
        response = self.post(
            **_prepare_payload(inputs, parameters, expect_binary=True),
            model=model,
            task="image-classification",
       )
```

(i.e. not a static method, `expect_binary=True` and using `**` on dict return value)